### PR TITLE
Battlesearch: Support searching for multiple users and searching months

### PR DESCRIPTION
**Pull request by mia-pi-git**
Various improvements requested by Dawoblefet. 
It now searches by month, not day, has a back button, supports searching for 2 users, and the formats list is reformatted.
Screenshots: multisearch (https://prnt.sc/uj8ih4), singlesearch (https://prnt.sc/uj8ioq)

### CODE REVIEW

**Comment by Zarel** [comment]
It doesn't seem to have any links to battle results. Is that just something we've never supported?

File: server/chat-plugins/modlog-viewer.ts [hunk 1]
```diff
@@ -20,12 +20,20 @@ import {Dex} from '../../sim/dex';
 import {Config} from '../config-loader';
 import {checkRipgrepAvailability, ModlogID} from '../modlog';
 
-interface Results {
-	[k: string]: number;
+interface BattleOutcome {
+	lost: string;
+	 won: string,
```
Comment (Zarel) [inline server/chat-plugins/modlog-viewer.ts:25]: Do you still not have `npm test` working?

**Comment by mia-pi-git** [comment]
> It doesn't seem to have any links to battle results. Is that just something we've never supported?

No, it just displays battle results as numbers in a total.

**Comment by DaWoblefet** [comment]
Typically matches aren't saved as replays for what we're concerned about anyway. So far as I know, battlesearch has only been used in cases of ladder boosting - we never use it to say, verify if there was harassment in a game that wasn't saved as a replay. That's only available to those with console access.

**Comment by Zarel** [comment]
Seems like it's unlikely to be a problem.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**